File: 백준/Silver/10828. 스택/스택.cc
#include<stack>
#include<iostream>
using namespace std;

int main() {
	stack<int> s1;
	int n;
	string k;
	cin >> n;
	for (int i = 0; i < n; i++) {
		int num = 0;
		cin >> k;
		if (k == "push") {
			cin >> num;
			s1.push(num);
		}
		else if (k == "top") {
			if (s1.empty()) cout << "-1" << "\n";
			else cout << s1.top() << "\n";
		}
		else if (k == "size") cout << s1.size() << "\n";
		else if (k == "empty") cout << s1.empty() << "\n";
		else if (k == "pop") {
			if (s1.empty()) cout << "-1"<<"\n";
			else {
				cout << s1.top() << "\n";
				s1.pop();
			}
		}
	}

}

File: 백준/Silver/10828. 스택/스택.py
import sys
n = int(input())
arr = []
for i in range(n):
    a = list(sys.stdin.readline().split())
    if a[0] == 'push':
        arr.append(int(a[1]))
    elif a[0] == 'pop':
        if not arr:
            print(-1)
        else:
            print(arr.pop())
    elif a[0] == 'size':
        print(len(arr))
    elif a[0] == 'empty':
        if arr:
            print(0)
        else:
            print(1)
    elif a[0] == 'top':
        if not arr:
            print(-1)
        else:
            print(arr[-1])
        